### ISSUE TITLE
please add orthography profile and conversion for segments to the csv pipelin

### ISSUE THREAD

**Issue by LinguList** [bambooforest/dogon#5]
Ideally, @bambooforest carries out the conversion right away, using orthography profiles, then I can jump in and look for the concepticon mapping. So please:

* add a column "segments"

**Comment by bambooforest** [comment]
do you need segments for concepticon mappings?

**Comment by LinguList** [comment]
nope, but we'll need them afterwards, so why not make them right away?

**Comment by bambooforest** [comment]
because right now if i add the segmentation routine it would apply to the full Dogon CSV file instead of just the subset that you are going to select :-)

**Comment by LinguList** [comment]
yes, this would be good! So we have potentially ALL data, right?

**Comment by bambooforest** [comment]
but if we're never going to use all the data for alignment, then there's no point. i've created the orthography profile to handle the cases that we found in the 300 word list. that took a lot of work because there are transcription systems from half a dozen people, and each differs slightly. i don't want to insure that all 8000 rows * 20+ languages are doing be handled by the OP if we're only going to use, say 3% of the data in the analysis

**Comment by LinguList** [comment]
then just please push the profile and some description how you handled those cases. I don't want to pass data around too many times, so preferably, I could just do all in a rush from there, but if you add the orthography profile, I'll be able to work on all things on the data.

I still think it would be better to test the orthoprofiles on the data for now. First, if it's transcription systems by different people, you could make it language-wise, right? second, we'd already know more or less, which data not to touch, etc., etc., as the things where the orthoprofiles throw an error could be kicked out, etc., etc., so maybe just try the things you HAVE on the data, and we'll see from then?

**Comment by LinguList** [comment]
and in any way: the next step should be to convert the spreadsheet into a wordlist-like file, so that I can do the concepticon-mapping. You did that last time, so this should also be done before I do anything on the concepts.

**Comment by bambooforest** [comment]
i don't know what a wordlist-like file is...

i do know that all the code and and orthography profile are already in this repository, e.g.

https://github.com/bambooforest/dogon/blob/master/working/Heath2014.prf

since you'd like everything in one place and in one pipeline, perhaps you could give me a list of exactly the steps you want to see and an idea of the output formats you required (i.e. a design spec) -- which all the parts are already in this repository, just in different directories, e.g.

https://github.com/bambooforest/dogon/tree/master/concepticon
https://github.com/bambooforest/dogon/tree/master/data
https://github.com/bambooforest/dogon/blob/master/data/final-wordlist.tsv

then when i have the design spec from you that says something like:

1. get the data from the spreadsheet via the Dogon website
2. convert it to csv
3. convert the data to lingpy format (? as per the final-wordlist.tsv file above? is that still the format you use?)
4. add a column with the segmented data
5. whatever else you need for your scripts

then i will implement these steps and i will delete all the other remaining directories and files in this repository (so if there's anything here you want to keep, i'd suggest making a local copy somewhere)

sound good? then i know exactly what you want, exactly how you want it, and then we don't have to change things down the road.

**Comment by LinguList** [comment]
please, I don't want to be forced to read through those things. You did not put the code of how you got from the csv original file to the file you gave me then, and I want you to do exactly that: convert the data into the list into which you converted it last time before I started working with it. And last time you WERE using your orthopgraphy profiles already in this stage, as I remember, but you never put the code online (at least I didn't see any Python code):

* https://github.com/bambooforest/dogon/blob/master/working/collapsed-dogon-wordlists.tsv

Now, we also want to trace the original data, so you should add another cell that traces the line number in the versionized CSV/XLSX file, and the column number, just to make sure one can go back rather easily, maybe as a tuple in one column (call it ORIGINAL_DATA, and values like 10,20 = line 10 cell 20).

**Comment by bambooforest** [comment]
so all you want is a transparent pipeline that produces this file?

https://github.com/bambooforest/dogon/blob/master/working/collapsed-dogon-wordlists.tsv

with an additional ID column that adds the cell number?

**Comment by LinguList** [comment]
cell number and row number, right!

Thsi is where I started last time, and I'd like to keep it that way as this is more economic (if we pass too much back and forth between the two of us, it will be difficult and cost more time).

**Comment by bambooforest** [comment]
do you need this "CONCEPT_LIST" in the final tsv file? i'd be hard pressed to recreate it because you're insisting on the dogon data from the dogon website, but to add things like the Leipzig Jakarata labels, i did this originally by hand matching.

**Comment by LinguList** [comment]
nope, I don't need them, as I know after matching to concepticon, right?
